cardano-node/test/Test/Cardano/Node/Json.hs

```diff
@@ -46,8 +49,14 @@ prop_roundtrip_NodeSetup_JSON :: Property
 prop_roundtrip_NodeSetup_JSON =
   Hedgehog.property $ do
     ns <- Hedgehog.forAll genNodeSetup
-    Hedgehog.tripping ns toJSON fromJSON
-    Hedgehog.tripping ns encode decode
+    Hedgehog.tripping ns toJSON (adjustUseLedger . fromJSON)
+    Hedgehog.tripping ns encode (adjustUseLedger . decode)
+  where
+    -- 'useLedger' will be parsed as 'Always' instead of 'After 0'
```

Comment:
That feels wrong. Why `toJSON` and `FromJSON` are different?

Reply:
Internally we parse `After 0` as `Always` and so in this case since the value is randomly generated we have to adjust it since the generator generates `After 0`

Reply:
Can this be fixed on generator level instead inside a condition?

Reply:
yes!